### ISSUE TITLE
Unexpected whitespace color.

### ISSUE THREAD

**Issue by dingyi1993** [aaron-bond/better-comments#7]
when I set : "editor.renderWhitespace": "all", whitespace in comment will be colored by this plugin.

**Comment by aaron-bond** [comment]
I've started to look at this behaviour. Not sure off-hand of a straightforward fix; it looks like the the `createTextEditorDecorationType` method doesn't take into account that you might want to leave the whitespace characters alone.

For reference, I've raised [this question on SO](https://stackoverflow.com/questions/45212977/vscode-extension-createtexteditordecorationtype-ignore-whitespace) just in case anyone out there knows about a sensible approach.

**Comment by hbergren** [comment]
Has there been any change to the `createTextEditorDecorationType` method signature, or any new developments in VS Code that would solve this issue?

I did a cursory glance and didn't find anything, but I haven't written a VS Code extension either, so I'm not sure what to look for.

**Comment by aaron-bond** [comment]
Hi @hbergren, I was looking into this issue again and the issue I found on VSCode's repo was the one you'd raised :)

As far as I can see, it's a stretch goal for the VSCode team to get this in for 2018, so I don't have high hopes.

If I wanted to implement this myself, I'd have to chop up the text ranges within the decorator to exclude whitespace - not a simple feat and probably killer on performance :(

**Comment by aaron-bond** [comment]
Pretty sure this is the feature I need before I can make this work: https://github.com/Microsoft/vscode/issues/585

**Comment by AshCoolman** [comment]
@aaron-bond perhaps unblocked?

> Done via https://github.com/microsoft/vscode/issues/86415

https://github.com/microsoft/vscode/issues/585#issuecomment-606280035

**Comment by alfonder** [comment]
2 years later still no solution?

**Comment by aaron-bond** [comment]
I've never heard anything back on my SO questions or other attempts to find a resolution to this.

I've raised a feature request here: https://github.com/microsoft/vscode/issues/102390
